extensions/statsd-emitter/src/main/java/io/druid/emitter/statsd/StatsDEmitter.java

```diff
@@ -0,0 +1,96 @@
+package io.druid.emitter.statsd;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSortedSet;
+import com.metamx.common.logger.Logger;
+import com.metamx.emitter.core.Emitter;
+import com.metamx.emitter.core.Event;
+import com.metamx.emitter.service.ServiceMetricEvent;
+import com.timgroup.statsd.NonBlockingStatsDClient;
+import com.timgroup.statsd.StatsDClient;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Created by michael.schiff on 2/5/16.
+ */
+public class StatsDEmitter implements Emitter
+{
+
+  private final static Logger log = new Logger(StatsDEmitter.class);
+
+  private final StatsDClient statsd;
+  private final Map<String, StatsDEmitterConfig.MetricType> metricTypes;
+
+  public StatsDEmitter(StatsDEmitterConfig config) {
+    statsd = new NonBlockingStatsDClient(config.getPrefix(), config.getHostname(), config.getPort());
+    metricTypes = config.getMetricTypes();
+  }
+
+
+  @Override
+  public void start()
+  {
+    log.info("Starting StatsD emitter");
+    /*
+      StatsD client is started when constructed.
+     */
+  }
+
+  @Override
+  public void emit(Event event)
+  {
+    if (event instanceof ServiceMetricEvent) {
+      ServiceMetricEvent metricEvent = (ServiceMetricEvent) event;
+      String host = metricEvent.getHost();
+      String service = metricEvent.getService();
+      String metric = metricEvent.getMetric();
+      Map<String, Object> userDims = metricEvent.getUserDims();
+      Number value = metricEvent.getValue();
+
+      ImmutableList.Builder<String> nameBuilder = new ImmutableList.Builder<>();
+      nameBuilder.add(service);
+      nameBuilder.add(metric);
+      ImmutableSortedSet<String> sortedUserDims = ImmutableSortedSet.copyOf(userDims.keySet());
+      for (Object dim : sortedUserDims) {
+        nameBuilder.add(userDims.get(dim).toString());
+      }
+      String fullName = Joiner.on(".").join(nameBuilder.build());
+
+      StatsDEmitterConfig.MetricType type = metricTypes.get(metric);
+
+      if (type != null) {
```

Comment:
My concern was that this could be a spammy log. I think if no StatsD metric type is specified, it makes sense not to send the metric. This should definitely be specified in the README though